### PR TITLE
1919 - moved filter response records to validation

### PR DESCRIPTION
**Pull request by slamothe99**
## Pre-review checklist

I have ensured that:

- [ ] Code runs successfully in the development environment.
- [ ] New code is tested to prove that it works. If not, justify reasoning.
- [ ] New code is documented using [type hinting](https://realpython.com/lessons/type-hinting/) and following the [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) docstring format.
- [ ] New code follows the project naming standards.

## Post-review checklist

As I merge these changes, I have:

- [ ] Updated the related Jira ticket, including approximate time spent (including review).
- [ ] Moved the related Jira ticket to Done.

### CODE REVIEW

**Review by foster999 — APPROVED** [review]

**Review by foster999 — CHANGES_REQUESTED** [review]
Please update the filtering to include the error from this check

**Review by ohibo — APPROVED** [review]

File: cishouseholds/validate_class.py [hunk 1]
```diff
@@ -26,6 +26,10 @@ def __init__(self, dataframe: DataFrame, error_column_name: str) -> None:
             "between": {"function": self.between, "error_message": "{} should be between {}{} and {}{}"},
             "null": {"function": self.not_null, "error_message": "{} should not be null"},
             "valid_vaccination": {"function": self.valid_vaccination, "error_message": "invalid vaccination"},
+            "valid_file_date": {
+                "function": self.check_valid_file_date,
+                "error_message": "file date should be after visit date",
```
Comment (foster999) [inline cishouseholds/validate_class.py:31]: We're checking the visit date in particular, so reverse wording focus to this. Also describe the rest of the check.

Please substitute in the actual column names using `{}`

```suggestion
                "error_message": "visit date should be before file date when samples have been collected",
```

File: cishouseholds/pipeline/validation_ETL.py [hunk 2]
```diff
@@ -105,7 +111,10 @@ def validation_ETL(df: DataFrame, validation_check_failure_column_name: str, dup
     SparkVal.count_complete_duplicates(duplicate_count_column_name)
     validation_calls(SparkVal)
     return SparkVal.filter(
-        selected_errors=["participant_id, visit_datetime, visit_id, ons_household_id should not be null"],
+        selected_errors=[
+            "participant_id, visit_datetime, visit_id, ons_household_id should not be null",
+            "the date in visit_datetime should be before the date expressed in survey_response_source_file and both swab_sample_barcode_column and blood_sample_barcode_column should be null",  # noqa:E501
```
Comment (foster999) [inline cishouseholds/pipeline/validation_ETL.py:116]:
```suggestion
            "the date in visit_datetime should be before the date expressed in survey_response_source_file when both swab_sample_barcode_column and blood_sample_barcode_column are null",  # noqa:E501
```

File: cishouseholds/validate_class.py [hunk 3]
```diff
@@ -26,6 +26,10 @@ def __init__(self, dataframe: DataFrame, error_column_name: str) -> None:
             "between": {"function": self.between, "error_message": "{} should be between {}{} and {}{}"},
             "null": {"function": self.not_null, "error_message": "{} should not be null"},
             "valid_vaccination": {"function": self.valid_vaccination, "error_message": "invalid vaccination"},
+            "valid_file_date": {
+                "function": self.check_valid_file_date,
+                "error_message": "the date in {} should be before the date expressed in {} and both {} and {} should be null",  # noqa:E501
```
Comment (foster999) [inline cishouseholds/validate_class.py:31]:
```suggestion
                "error_message": "the date in {} should be before the date expressed in {} when both {} and {} are null",  # noqa:E501
```

**Review by foster999 — APPROVED** [review]

**Pull request merged**